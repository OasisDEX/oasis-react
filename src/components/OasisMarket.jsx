import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";

import styles from "./OasisMarket.scss";
import CSSModules from "react-css-modules";
import { createEtherscanAddressLink } from "../utils/createEtherscanTransactionLink";

const propTypes = PropTypes && {
  marketAddress: PropTypes.string,
  networkName: PropTypes.string
};
const defaultProps = {};

export class OasisMarket extends PureComponent {
  render() {
    const { marketAddress, networkName } = this.props;
    const etherscanLink = createEtherscanAddressLink({
      networkName,
      address: marketAddress
    });
    return (
      <div styleName="OasisMarket">
        <span styleName="label"> Market: </span>
        <div styleName="OasisMarketAddress">
          <a target="_blank" rel="noopener noreferrer" href={etherscanLink}>
            {marketAddress}
          </a>
        </div>
      </div>
    );
  }
}

OasisMarket.displayName = "OasisMarket";
OasisMarket.propTypes = propTypes;
OasisMarket.defaultProps = defaultProps;
export default CSSModules(OasisMarket, styles);
