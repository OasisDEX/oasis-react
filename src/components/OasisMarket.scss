@import "../styles/modules/breakpoints";
@import "../styles/modules/typography";

.OasisMarket {
  @include text(13px);
  text-align: right;
  & .OasisMarketAddress {
    border: 1px solid rgba(211, 213, 220, 0.68);
    border-radius: 100px;
    font-size: 10px;
    background-color: transparent;
    margin-left: 25px;
    display: inline-flex;
    padding: 0 17px 0 10px;
    width: 330px;
    @media (max-width: $xxs) {
      margin-left: 0;
      align-content: flex-start;
      width: 100%;
      padding-right: 0;
    }


    & * {
      color: #D0D0D0;
      transform: translateX(-10px);
      padding: 13px 17px;
      display: inline-block;
      font-family: 'Montserrat', sans-serif;
      @media (max-width: $xxs) {
        @include text(9px);
      }
      -moz-user-select: none;
      -webkit-user-select: none;
      -ms-user-select: none;
      user-select: none;
    }

    & a {
      padding-left: 45px;
      max-width: 100%;
      @media (max-width: $xxs) {
        padding-left: 87px;
        letter-spacing: 0.25px;
      }
    }

  }
    & > .label {
      @media (max-width: $xxs) {
        position: absolute;
        top: 9px;
        letter-spacing: 1.3px;
        left: 30px;
      }
    }

}
