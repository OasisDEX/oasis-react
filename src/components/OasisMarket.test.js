/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import { OasisMarket } from './OasisMarket';
import { shallow } from 'enzyme';

describe('(Component) OasisMarket', () => {
  it('should render', () => {
    const props = {};
    const wrapper = shallow(
      <OasisMarket {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
