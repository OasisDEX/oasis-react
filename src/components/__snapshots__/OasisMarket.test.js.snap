// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisMarket should render 1`] = `
<div
  styleName="OasisMarket"
>
  <span
    styleName="label"
  >
     Market: 
  </span>
  <div
    styleName="OasisMarketAddress"
  >
    <a
      href="https://kovan.etherscan.io/address/0x1234"
      rel="noopener noreferrer"
      target="_blank"
    >
      0x1234
    </a>
  </div>
</div>
`;
